CONTRIBUTING.md

```diff
@@ -0,0 +1,96 @@
+ * Intend your code with taps rather than 4 spaces.
+
+### External Contributions
+
+If you want to provide a new feature, a bug-fix, please follow these steps:
+
+0) Fork this repository
+1) Create a branch named exactly as the issue you are working on. If the issue has the label "urgent", branch from dev, else branch from NIGHTLY.    
+2) If you want to develop a new feature: Design your code. A simple class-diagram is enough, but if you go the extra mile to also provide a sequenz-diagramm, this wont hurt either.
+3) Fix the bug/develop the new feature based on your design.
+4) Write a test. Test any case you could come up with.
+5) Open a pull-request based on the provided pull-request template onto the branch your branch branched from.
+6) If your pull-request get's denied, start over from 3.
+
+**NOTE**: Only change thing relevant to your issue! If your pull-request changes anything, that is not important, your pull-request will be denied!
+
+If you want to contribute a new feature to NetCom2, you should have extensive knowledge about NetCom2 and Client-Server-Communications.
+
+In most cases, pull-request from external contributors about urgent Issues or new features will be denied.
+
+### Direct Contributions
+
+If you are a direct contributor (you have push rights to this repository), please follow the following guid
+
+1) Create a branch named exactly as the issue you are working on. If the issue has the label "urgent", branch from dev, else branch from NIGHTLY.    
+2) If you want to develop a new feature: Design your code. A simple class-diagram is enough, but if you go the extra mile to also provide a sequenz-diagramm, this wont hurt either.
+3) Fix the bug/develop the new feature based on your design.
+4) Write a test. Test any case you could come up with.
+5) Open a pull-request based on the provided pull-request template onto the branch your branch branched from.
+6) If your pull-request get's denied, start over from 3.
+
+### Releases
+
+The right to provide a new release lies by the repository owner.
+
+### Becoming a direct Contributor
+
+For becoming a direct contributor, you have to apply to the owner of this repository.
+
+You should already have:
+
+ * Extensive knowledge about software-design
+ * Fluent in UML
+ * Fluent in Java
+ * Extensive knowledge about Client-Server-Communication
+ * Optional: Done previous bug-fixes in this repository
+ 
+You are granted:
+ 
+ * The right to push.
+ * The right to speak in behalf of NetCom2.
+ * The right to alter the existing templates (requires the acknowledge and the okay from the repository owner)
+ * Whatever you don't know, you will be taught. (The repository-owner does not have all the time in the world, so please resign to any direct contributor for help)
+ 
+The repository owner holds a local copy of the project. If you do any malicious action or an major break, the repository owner holds his right to remove you as a direct contributor.
```

Comment:
"The repository owner keeps a local copy of the project. If you perform any malicious action or cause a major break, the repository owner reserves the right to remove you as a direct contributor."